doc/level1.rst

```diff
@@ -249,40 +249,55 @@ Internal state storage
 
-.. todo::
-
-   Explain why we keep peer state.  because we want to handle weird cases like the following...
-
-If a remote peer disables Autocrypt or drops back to using a
-non-Autocrypt MUA only we must be able to disable sending encrypted
-mails to this peer automatically.  MUAs capable of Autocrypt Level 1
-therefore MUST store state about the capabilities of their remote
-peers.
+Autocrypt clients keep state about their peers, to be able to handle
+several nuanced situations that have caused trouble/annoyance in the
+past.  This state is updated even when the peer does *not* send mail
+with an ``Autocrypt`` header.
 
-Agents MAY also store additional information gathered for heuristic
-purposes, or for other cryptographic schemes.  However, in order to
-support future syncing of Autocrypt state between agents, it is
-critical that Autocrypt-capable agents maintain the state specified
-here.
+For example, if a remote peer disables Autocrypt or drops back to
+using a non-Autocrypt MUA only we must be able to disable sending
+encrypted mails to this peer automatically.
 
-.. todo::
+In addition to the per-peer state described in :ref:`peer-state`,
+agents MAY also store other information gathered for heuristic
+purposes, or for other cryptographic schemes (see
+:doc:`optional-state` for some example ideas).
 
-   Explain the values for each of the known ``state`` options:
-   ``nopreference``, ``mutual``, ``reset``, and ``gossip``.
+However, in order to support future syncing of Autocrypt state between
+agents, it is critical that Autocrypt-capable agents maintain the
+state specified here, regardless of what additional state they track.
 
 .. note::
 
-  - The above is not necessarily an exhaustive list of peer state to
-    keep; implementors are encouraged to improve upon this scheme as
-    they see fit. Suggestions for additional (optional) state that an
-    agent may want to keep about a peer can be found in
-    :doc:`optional-state`.
   - An implementation MAY also choose to use keys from other sources
     (e.g. local keyring) at own discretion.
-  - If an implementation chooses to automatically ingest keys from a
-    ``application/pgp-keys`` attachment, it should only do so if they
-    have a matching user id.
+  - If an implementation chooses to automatically ingest a key from a
+    ``application/pgp-keys`` attachment as though it was found in an
+    ``Autocrypt`` header, it should only do so if the attached key has
+    a :rfc:`User ID <4880#section-5.11>` which matches the message's
+    ``From`` address.
+
+``peer_state[A].state`` semantics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The ``state`` variable of a particular peer's ``peer_state`` data is
+selected from a set range of values:
+
+  - ``nopreference`` means the peer has not opted into mutual
+    encryption.  The client may or may not know a key for such a peer.
+  - ``mutual`` means we know a key for the peer, and the peer has also
+    opted into mutual encryption.
```

Comment:
"mutual encryption" by itself does not quite cover this. I think this is more clearly phrased somewhere else, something like "expressed agreement to encrypt by default if all parties consent".